### PR TITLE
Add native node package manager with resolve support

### PR DESCRIPTION
**Pull request by mattcompiles**
# ↪️ Pull Request

This PR adds a native node package manager for use in the parcel config request. It also changes the node resolver implementation to use a `FileSystemRef` type rather than a FileSystem generic. 

Happy to add any more tests where we feel is appropriate.

### CODE REVIEW

File: crates/parcel_config/src/config_error.rs [hunk 1]
```diff
@@ -26,6 +26,6 @@ pub enum ConfigError {
     from: PathBuf,
     specifier: String,
     #[source]
-    source: Box<dyn std::error::Error>,
+    source: Box<anyhow::Error>,
```
Comment (yamadapc) [inline crates/parcel_config/src/config_error.rs]: you can unbox it

**Review by yamadapc — APPROVED** [review]

File: crates/parcel_package_manager/src/node_package_manager.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      ),
+    }
+  }
+}
+
+impl<'a> PackageManager for NodePackageManager<'a> {
+  fn resolve(&self, specifier: &str, from: &std::path::Path) -> anyhow::Result<crate::Resolution> {
+    let res = self.resolver.resolve(specifier, from, SpecifierType::Cjs);
+
+    match res.result? {
```
Comment (yamadapc) [inline crates/parcel_package_manager/src/node_package_manager.rs:28]: for what it's worth you could write this just with `??`

```
let (Resolution::Path(pathbuf), _) = res.result?.map_err(|err| anyhow!(...))?;
Ok(...)
```

**Review by MonicaOlejniczak — APPROVED** [review]

**Comment by devongovett** [comment]
This is modifying the existing resolver which is used in v2 today. So I guess now would be the time to double check that moving to dyn doesn't have any performance impact.

**Comment by mattcompiles** [comment]
@devongovett I tested this across a bunch of Jira builds are I couldn't detect any noticeable difference in performance.

**Review by MonicaOlejniczak — APPROVED** [review]

File: crates/node-bindings/src/resolver.rs [hunk 3]
```diff
@@ -233,38 +192,38 @@ pub struct JsInvalidations {
 #[napi]
 pub struct Resolver {
   mode: u8,
-  #[cfg(not(target_arch = "wasm32"))]
-  resolver: parcel_resolver::Resolver<'static, EitherFs<JsFileSystem, OsFileSystem>>,
-  #[cfg(target_arch = "wasm32")]
-  resolver: parcel_resolver::Resolver<'static, JsFileSystem>,
+  resolver: parcel_resolver::Resolver<'static>,
   #[cfg(not(target_arch = "wasm32"))]
   invalidations_cache: parcel_dev_dep_resolver::Cache,
```
Comment (alshdavid) [inline crates/node-bindings/src/resolver.rs:197]: Does this need to be conditionally compiled out? Can we supply an empty `Cache` for `wasm` builds?

**Review by yamadapc — APPROVED** [review]

**Review by alshdavid — APPROVED** [review]

File: crates/node-bindings/src/resolver.rs (hunk 3 above)
Comment (mattcompiles) [reply crates/node-bindings/src/resolver.rs:197]: My change actually removes the need for some of the conditions. We can revisit the other stuff later as it's not related to my changes.

**Pull request merged**